packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1012,8 +1012,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         raiseConnectedEvent(this._logger, this, connected, clientId);
 
         if (connected) {
-            assert(!!clientId);
-            this.summaryManager.setConnected(clientId);
+            if (clientId) {
```

Comment:
why do we need this code? I.e. if you provide non-empty clientID, does container runtime needs to know anything about this new state? I believe the answer is no, it should behave exactly as connection to a service where user has only read access to a file (i.e. can not modify it).
Other way to put it - do we need to leak that new state to container runtime?